src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/AdditionalDesignTimeBuildInput.xaml

```diff
@@ -0,0 +1,23 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
+<Rule Name="AdditionalDesignTimeBuildInput"
+      PageTemplate="generic"
+      xmlns="http://schemas.microsoft.com/build/2009/properties">
+  <Rule.DataSource>
+    <DataSource HasConfigurationCondition="False"
+                ItemType="AdditionalDesignTimeBuildInput"
+                Persistence="ProjectFile"
+                SourceOfDefaultValue="AfterContext" />
+  </Rule.DataSource>
+
+  <StringProperty Name="FullPath"
+                  ReadOnly="true"
+                  Visible="false">
+    <StringProperty.DataSource>
+      <DataSource ItemType="AdditionalDesignTimeBuildInput"
```

Comment:
An aside: I haven't looked much into per-property data sources. Can you think of validation rules we can apply on them? Do they behave the same way as top-level rule data sources? My guess was not, so #5000 only validates the top level occurrences.

Reply:
I've respecified the values, it would be good to know if these need to be respecified, and either way enforced. They have at least one additional rule, that ItemType needs to match the parent data source ItemType.